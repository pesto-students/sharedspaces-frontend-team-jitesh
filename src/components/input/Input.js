import React from 'react'

const Input = ({ name, value, type, id, placeholder, label, onChange, required }) => {
    return (
        <div className="input-group mb-4">
            <label htmlFor={name}>{label}</label>
            <input
                onChange={(e) => onChange(e)}
                name={name}
                value={value}
                type={type}
                id={id}
                placeholder={placeholder}
                className='border-2 px-2 py-1.5 rounded transition duration-150 ease-in-out focus:border-red-500'
                required
            />
        </div>
    )
}

export default Input