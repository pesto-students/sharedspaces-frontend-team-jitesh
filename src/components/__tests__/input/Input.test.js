import { render, screen, cleanup } from "@testing-library/react";
import renderer from 'react-test-renderer'
import Input from '../../input/Input'

afterEach(() => {
    cleanup()
})

test('test matches snapshot', () => {
    const component = renderer.create(<Input />)

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot();
})


test('should render input component', () => {
    render(<Input />)

    const inputElement = screen.getByTestId("input")

    expect(inputElement).toBeInTheDocument();
})