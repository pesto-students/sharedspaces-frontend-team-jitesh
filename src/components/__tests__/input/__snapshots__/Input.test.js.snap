// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test matches snapshot 1`] = `
<div
  className="input-group mb-4 undefined"
  data-testid="input"
>
  <label />
  <input
    className="border-2 px-2 py-1.5 rounded transition duration-150 ease-in-out focus:border-gray-500"
    disabled={false}
    onChange={[Function]}
    required={true}
  />
</div>
`;
